### ISSUE TITLE
Accessibility utility class for hiding elements but making it visible only for screenreaders, not breakpoint specific.

### ISSUE THREAD

**Issue by mfrejnik** [zurb/foundation#6366]
I know Foundation has utility classes for that. However they all are resolution related like for eg `hidden-for-small-only`
But I would like to have one class to just make the element screenreader only for all breakpoints.
Something like `hidden-for-all`.
What do you think?

**Comment by Lynesth** [comment]
Hey there.

There actually is an undocument class named "hidden-for-small-up" which will hide it on all resolutions.

Regards,

Lyn.

**Comment by mfrejnik** [comment]
Thanks a lot for the answer. So it basically means that this "issue" should be either clarified in documentation or be rewritten => simplified to one class to hide for all resolution

**Comment by gakimball** [comment]
We're actually wondering that right now for Foundation 6. While we were doing the accessibility push we ended up with a slew of new accessibility-focused visibility classes, which also have sizing classes (`visible-for-small`, `hidden-for-medium`). But how necessary are those, really? If you're using AT to read a web page, it seems like the screen size wouldn't really matter; just the text.

If that's the case, there are two basic scenarios you'd want:
- Not visible on screen readers, by using `aria-hidden="true"`
- Only visible on screen readers, by using a CSS class that does that clipping trick

Does that make sense, or do we need all of these responsive accessibility sizing classes? They add a lot of weight to the CSS.

**Comment by Owlbertz** [comment]
I agree on that it is rather not usual to have the need for size-based accessibility. Personally, I'm always good with one (either the .visuallyhidden from HTML5 boilerplate or the .hidden-for-small-up, which Lynesth mentioned). Tbh, I currently prefer adding the .visuallyhidden to my CSS, since the other one is pretty inconvenient. So from my point of view we're good with one class for all sizes, with a shorter name like just .hidden.

**Comment by gakimball** [comment]
That's what I thought. Right now in our Foundation 6 alpha we have just two helper classes:
- `.hide-for-sr`: visually hides content while allowing AT to read it. It uses the `element-invisible()` mixin that Foundation 5 already has.
- `.show-on-focus`: hides content unless it has focus, for creating skip links.

No idea how many people are actually using the current accessibility visibility classes, but we can't really remove them since that's a breaking change. We could add the above two classes for now as a stopgap, though.

**Comment by Lynesth** [comment]
@gakimball : I agree that a simple ```.hide-for-sr``` will be enough but if it's going to be in Foundation 6, the increase in version number should allow the removal of useless classes. People will not just upgrade their projects which are already in prod from 5 to 6.

I mean, that's what a major increase in version means : that some stuff are not BC.

**Comment by gakimball** [comment]
@Lynesth Yeah, we're being *really* aggressive with how we trim the codebase going from 5 to 6. The upgrade path is going to be more difficult than it was going from 4 to 5, but the codebase really needs it.

**Comment by mfrejnik** [comment]
That absolute make sense @Lynesth I was actually looking for something simple like one class `.hide-for-sr`. Removing the legacy with size dependent accessibility classes is way to go with Foundation V6. Can't wait for the new V6.
